tests/compiler/compare-null.ts

```diff
@@ -0,0 +1,51 @@
+class CompareNullA {}
+
+export function compare_null_with_null(): void {
+  assert(null == null);
+  assert(!(null != null));
+}
+compare_null_with_null();
+
+export function compare_null_with_integer(): void {
+  assert(null == 0);
+  assert(!(null != 0));
+  assert(!(null == 10));
+  assert(null != 10);
+}
+compare_null_with_integer();
+
+export function compare_null_with_operator_overload(): void {
+  null == "aa";
+  null != "aa";
+  "aa" == null;
+  "aa" != null;
+
+  null === "aa";
+  null !== "aa";
+  "aa" === null;
+  "aa" !== null;
+}
```

Comment:
This is something to be very careful with. Your current code is perfectly fine for String, but I have to wonder what happens with classes with operator overloads that have signatures with non-null parameters. In that case, you would need to add some null-checking code for `==` that's equivalent to `a && b ? eqOverload(a, b) : a == b`, where `a` and `b` are the pointers of the LHS and RHS respectively, and `eqOverload` represents the overload function. Similar null-checking code applies to `!=`. You could also opt to error if `==`/`!=` is used on nullable types that have an overload without nullable parameters.